src/pin.rs

```diff
@@ -0,0 +1,219 @@
+//! Mock digital InputPin and OutputPin implementations
```

Comment:
Maybe link to embedded-hal docs?

```diff
--- a/src/pin.rs
+++ b/src/pin.rs
@@ -1,4 +1,7 @@
-//! Mock digital InputPin and OutputPin implementations
+//! Mock digital [`InputPin`] and [`OutputPin`] implementations
+//!
+//! [`InputPin`]: https://docs.rs/embedded-hal/latest/embedded_hal/digital/trait.InputPin.html
+//! [`OutputPin`]: https://docs.rs/embedded-hal/latest/embedded_hal/digital/trait.OutputPin.html
```

(Unfortunately [RFC1946](https://rust-lang.github.io/rfcs/1946-intra-rustdoc-links.html) is not yet stabilized...)